examples/computer_vision/_mnist_pl/model_def.py

```diff
@@ -0,0 +1,26 @@
+"""
+This example shows how to interact with the Determined PyTorch Lightning Adapter
+interface to build a basic MNIST network. The PLAdapter utilizes the provided
+LightningModule with Determined's PyTorch control loop.
+"""
+
+from determined.pytorch import PyTorchTrial, PyTorchTrialContext, DataLoader
+from determined.pytorch._lightning import PLAdapter
+import mnist
+
+class MNISTTrial(PLAdapter):
+    def __init__(self, context: PyTorchTrialContext) -> None:
+        lm = mnist.LightningMNISTClassifier(lr=context.get_hparam('learning_rate'))
+        self.dm = mnist.MNISTDataModule()
+
+        super().__init__(context, lightning_module=lm)
+
+    def build_training_data_loader(self) -> DataLoader:
+        self.dm.setup()
```

Comment:
This doesn't seem to be right since this function downloads data to the same directory for all the GPUs, which might cause errors

Reply:
I moved the download section to dm.prepare() as PL suggests and used local_rank to point it to avoid collisions